packages/web-components/fast-foundation/src/select/select.ts

```diff
@@ -0,0 +1,224 @@
+        super.valueChanged(previous, next);
+        DOM.queueUpdate(() => {
+            const selectedOption = this.listbox.selectedOption;
+            this.displayValue = selectedOption.textContent || selectedOption.value;
+        });
+    }
+
+    private updateForm(): void {
+        const value = this.value ? this.value : null;
+        this.setFormValue(value, value);
+    }
+
+    public keydownHandler = (e: KeyboardEvent): void => {
+        super.keypressHandler(e);
+
+        const keyCode = e.key || e.key.charCodeAt(0);
+
+        if (keyCode === "Escape") {
+            this.open = false;
+            DOM.queueUpdate(() => this.button.focus());
+            return;
+        }
+
+        if ([" ", "Enter"].includes(`${keyCode}`)) {
+            e.preventDefault();
+            e.stopPropagation();
+
+            if (!this.disabled) {
+                this.open = !this.open;
+            }
+
+            if (this.open) {
+                this.listbox.focusFirstSelectedOption();
+            } else {
+                DOM.queueUpdate(() => this.button.focus());
+            }
+
+            return;
+        }
+
+        // Pass the event to the listbox
+        if (!e.defaultPrevented) {
+            DOM.queueUpdate(() => {
+                e.preventDefault();
+                this.listbox.dispatchEvent(e);
+            });
+        }
+    };
+
+    public focusoutHandler = (e: FocusEvent): void => {
+        const focusTarget = e.relatedTarget as HTMLElement;
+
+        if (!this.open) {
+            return;
+        }
+
+        if (this.isSameNode(focusTarget)) {
+            e.stopPropagation();
+            e.preventDefault();
+            return;
+        }
+
+        if (!this.listbox.listboxItems.includes(focusTarget as Option)) {
+            e.stopPropagation();
+            this.open = false;
+
+            if (!focusTarget) {
+                DOM.queueUpdate(() => this.button.focus());
+            }
+        }
+    }
+
+    public clickHandler = (e: MouseEvent): void => {
+        if (this.disabled) {
+            return;
+        }
+
+        this.open = !this.open;
+
+        if (this.open) {
+            this.listbox.focusFirstSelectedOption();
+            return;
+        }
+
+        DOM.queueUpdate(() => this.button.focus());
+    };
+
+    @observable
+    public selectedValue: HTMLElement;
+    /**
+     * This will update the text that is in the select's
+     * button by default that renders the selected value
+     *
+     * @param value - This is the value for the <option>
+     */
+    private updateSelectValue(value: string) {
+        this.value = value;
+        if (this.selectedValue) {
+            this.selectedValue.textContent = value;
+        }
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        this.options = this.listbox.listboxItems;
+
+        this.button.setAttribute("aria-haspopup", "listbox");
```

Comment:
Is there a reason these are added imperatively in the connectedCallback and not declaratively in the template?

Reply:
The convention in the early draft added these imperatively, stating that they'd need to be added if the default slotted content was replaced. I wasn't sure so I left them as is, but I prefer to have them in the template instead.

Reply:
It doesn't look like it's in a slot anymore so I think it is probably safe to move to the template.